erpnext/accounts/doctype/sales_invoice/sales_invoice.py

```diff
@@ -852,7 +859,7 @@ def make_write_off_gl_entry(self, gl_entries):
 					"debit": self.base_write_off_amount,
 					"debit_in_account_currency": self.base_write_off_amount \
 						if write_off_account_currency==self.company_currency else self.write_off_amount,
-					"cost_center": self.write_off_cost_center or default_cost_center
+					"cost_center": self.cost_center or self.write_off_cost_center or default_cost_center
```

Comment:
`self.write_off_cost_center or or self.cost_center or default_cost_center`

Reply:
If user opted to tag cost center in Balance Sheet Accounts then write-off amount should also go to Cost Center selected in parent document otherwise trial balance wouldn't match.

Reply:
ok